### ISSUE TITLE
Empty .scalafmt.conf fails to parse

### ISSUE THREAD

**Issue by reid-spencer** [scalameta/scalafmt#991]
- **Version**: 1.0.0-RC4 
- **Integration**: IntelliJ Plugin 
- **IntelliJ Version**: 
```IntelliJ IDEA 2017.1.4, Build #IU-171.4694.23, built on June 6, 2017
JRE: 1.8.0_112-release-736-b21 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-79-generic
```
- **Configuration**: 
```
<Any .scalafmt.conf file content, including empty>
```

## Steps

* Type Ctrl-Shft-L or select "Reformat with scalafmt" from "Code" menu

## Problem
The scalafmt plugin generates this message in the IJ Event Log window:
```
8:48 AM	Scalafmt
					Failed to read .scalafmt.conf. 
					Invalid fields: callSite, defnSite, lambdaParameter
```

Note that none of those three strings appear in the .scalafmt.conf file.

## Expectation

I would like scalafmt to be able to read its input, even empty input, and proceed with formatting the file as requested. 

## Workaround

* Close all other projects that have a .scalafmt.conf file in them so that the only project open is the one with the file you're trying to format. 

## Notes

* This seems to be a problem with reading the *wrong* .scalafmt.conf file when multiple projects are open.
* I didn't see this problem in prior version of the IJ plugin (before 1.0.0 release candidates).

**Comment by olafurpg** [comment]
Thank you for reporting! There is no RC4 release for the intellij plugin, do you mean RC3?

**Comment by reid-spencer** [comment]
Yes, I do. Sorry, that was a typo or thinko.

**Comment by reid-spencer** [comment]
I have a few more details to share. This seems to be a situation where IntelliJ has two projects open that are in nested subdirectories. For example, suppose we have project P (parent) and project C (child) that are arranged in the file system like .../P/C -- that is, project C is checked out as a subdirectory of project P. I believe this is the scenario that causes the problem because when I correct the .scalafmt.conf in project P, I don't get the problem in project P any more. 

Just an idea, but if the algorithm to find a .scalafmt.conf traverses project C's path elements, it is going to find project P's first. You want to start from the root the of current project, not from the root of the file system. 

The title is probably wrong and should be changed to: "Nested projects cause wrong .scalafmt.conf to be read"

**Comment by olafurpg** [comment]
I'm curious to know if the fixes in #1007 help address this issue. 1.2.0 on the nightly channels is already released https://plugins.jetbrains.com/plugin/8236-scalafmt with that fix and 1.2.0 will come as soon

**Comment by reid-spencer** [comment]
When scalafmt updates in my IDEA I will check it out.

**Comment by reid-spencer** [comment]
No, it doesn't fix the problem. I'm running 1.2.0 in my IDEA now and the same problem exists. It uses the formatting in the parent project, not (sub-)project of the file being formatted.

**Comment by olafurpg** [comment]
Aww, shoot. If you're feeling ambitious, you can try to use https://github.com/scalameta/scalafmt/pull/1007 as inspiration to fix the problem. Contributing setup guide is here https://github.com/scalameta/scalafmt/blob/master/scalafmt-intellij/readme.md

**Comment by reid-spencer** [comment]
Probably won't have time to look at fixing the problem, but I will, eventually produce a simple reproduction case as a github repository that you can test it on .. assuming you still need that.

**Comment by olafurpg** [comment]
A minimal repro is definitely the best way to start. I probably won't have time to look into this personally, but maybe someone else who is also affected by this issue will.

**Comment by steinar** [comment]
Got a similar error after updating the scalafmt plugin:
```
Failed to read [...].scalafmt.conf. Invalid fields: callSite, defnSite
```

Resovled by removing the following deprecated parameters from `.scalafmt.conf`:
```
binPack.callSite = false
binPack.defnSite = false
```

Note: IntelliJ may not pick up the updated config file immediately.

**Comment by olafurpg** [comment]
Builtin support for scalafmt is now in the latest IntelliJ Scala EAP. Please try it out to see if it picks up the configuration as you expect it to

![screen shot 2018-07-16 at 16 58 45](https://user-images.githubusercontent.com/1408093/42767923-7daf4b6e-891e-11e8-9a40-23e89c67f80a.png)

**Comment by reid-spencer** [comment]
Will do when I get home.
